### PR TITLE
Issue #12827: Update doc for SuppressionCommentFilter

### PR DESCRIPTION
**Pull request by shubh220922**
Related to #12827

### CODE REVIEW

**Comment by romani** [comment]
GitHub, generate website

**Comment by nrmancuso** [comment]
> GitHub, generate website

@romani 

Github, generate site

**Comment by shubh220922** [comment]
GitHub, generate website

**Comment by shubh220922** [comment]
Is github not generating website because of CI failures?

**Comment by nrmancuso** [comment]
Github, generate site

**Comment by romani** [comment]
@shubh220922, CI must be green

**Comment by shubh220922** [comment]
GitHub, generate website

**Comment by shubh220922** [comment]
Github, generate website

**Comment by shubh220922** [comment]
@romani @nrmancuso Done. Please see https://github.com/checkstyle/checkstyle/pull/12836#issuecomment-1469454416

**Comment by romani** [comment]
@shubh220922 , please fix spelling, see CI failures.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java:174]: do indenation as +2 , not a +4.
we need more horizontal space.

File: src/xdocs/config_filters.xml
Comment (romani) [inline src/xdocs/config_filters.xml:252]: keep property indentation as +2.

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Comment by nrmancuso** [comment]
@shubh220922 we are not able to see your review on our end, you must submit it for others to see. Note “pending” status that is probably on your items.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java:250]: >ILLEGAL OFF: Exception


please some different var value `ILLEGAL OFF: Exception` to show that it tries to suppress by Check Name + message, so mismatch by catch type matters.
```
try {}
catch(Exception exc) {} // suppressed
catch(Error exc) {} // violation
```

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
Comment (shubh220922) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java]: Done.

File: src/xdocs/config_filters.xml
Comment (shubh220922) [reply src/xdocs/config_filters.xml]: Done.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
Comment (shubh220922) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java]: Done.

**Comment by shubh220922** [comment]
CI failed for spellchecker (Fixed).
image
(Image posted for benefit of new contributors)

**Comment by shubh220922** [comment]
@romani i haven't included examples for property ``checkC`` and ``checkCPP``. Should I include that? I wasn't there in the original documentation.

**Comment by shubh220922** [comment]
All this time I had added reviews on my code waiting for reply thinking it was visible to maintainers but it seems my reviews with ``pending`` status is only visible to me. :_(

**Comment by shubh220922** [comment]
Github, generate website

**Comment by romani** [comment]
Please all others, we try to keep same code in all examples to let user quickly see a difference in behavior.

**Comment by shubh220922** [comment]
@romani examples have been made consistent.

**Comment by shubh220922** [comment]
Github, generate website

**Comment by shubh220922** [comment]
Github, generate website

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java:141]: Fix indentation, it should be +2 for all

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
Comment (shubh220922) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java]: Done.

**Comment by shubh220922** [comment]
Github, generate website

**Comment by shubh220922** [comment]
Github, generate website

**Comment by shubh220922** [comment]
@romani 

CLI output >

```shell
PS C:\Users\shubh\OneDrive\Desktop\checkstyle project> cat config2.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="severity" value="error"/>
    <property name="fileExtensions" value="java, properties, xml"/>
    <module name="TreeWalker">
        <module name="SuppressionCommentFilter"/>
        <module name="MemberName"/>
        <module name="ConstantName"/>
        <module name="IllegalCatch"/>
    </module>
</module>
PS C:\Users\shubh\OneDrive\Desktop\checkstyle project> cat test2.java
class InputSuppressionCommentFilter
{
  int VAR1; // violation , Name 'VAR1' must match pattern '^[a-z][a-zA-Z0-9]*$'

  //CHECKSTYLE:OFF
  int VAR2; // suppressed violation
  //CHECKSTYLE:ON

  public static final int var3;
  // violation above , Name 'var3' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'

  //CHECKSTYLE:OFF
  public static final int var4; // suppressed violation
  //CHECKSTYLE:ON

  public void method1() {

  try{

  }catch(Exception ex){ // violation , Catching 'Exception' is not allowed

  }

  //CHECKSTYLE:OFF
  try{

  //CHECKSTYLE:ON
  }
}
PS C:\Users\shubh\OneDrive\Desktop\checkstyle project> java -jar "checkstyle-10.7.0-all.jar" -c config2.xml test2.java
Starting audit...
[ERROR] C:\Users\shubh\OneDrive\Desktop\checkstyle project\test2.java:3:7: Name 'VAR1' must match pattern '^[a-z][a-zA-Z0-9]*$'. [MemberName]
[ERROR] C:\Users\shubh\OneDrive\Desktop\checkstyle project\test2.java:9:27: Name 'var3' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [ConstantName]
[ERROR] C:\Users\shubh\OneDrive\Desktop\checkstyle project\test2.java:20:4: Catching 'Exception' is not allowed. [IllegalCatch]
Audit done.
Checkstyle ends with 3 errors.
PS C:\Users\shubh\OneDrive\Desktop\checkstyle project>

```

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java:141]: please respect indentation in method content, it should be +2.

without proper indentation it is hard to read example, border of method is not visible.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java:427]: `Example of how to configure the check to suppress Checks by name defined in comment.`

**Review by romani — CHANGES_REQUESTED** [review]
last two items:

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
Comment (shubh220922) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java:141]: @romani could you please be specific. I have kept 2 spaces after ``*`` in the javadoc. Is indentation required elsewhere. 

> border of method is not visible.

Do you want me to keep ``{`` and ``}`` at the same indentation level?

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java:141]: Yes

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
Comment (shubh220922) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java]: Done

**Comment by shubh220922** [comment]
@romani should I share the config again.

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
GitHub, generate website

**Comment by romani** [comment]
@shubh220922 , awesome, but I noticed that checkCPP,checkC properties are not covered.
can you update existing configs and example code to show them in action ?

**Comment by shubh220922** [comment]
@romani 

that was addressed here https://github.com/checkstyle/checkstyle/pull/12836#issuecomment-1475715947
Initially I didn't know how to write multiline comment inside a javadoc comment as writing the ``*/`` would terminate the javadoc itself. So I for some reason presumed it was not possible to write multiline comment at all ``/*.....*/`` as it was also not included in the original documentation but i did some searching and I will implement the below workaround for inserting multiline comment. 

```java
/**
 * This is a Javadoc comment for the exampleMethod.
 *
 * <pre>
 * {@code
 *   // This is a multiline comment-like section
 *   // inside the Javadoc comment.
 *   // You can also include code snippets here:
 *   int example = 10;
 *   System.out.println(example);
 * }
 * </pre>
 *
 * @param value an example parameter
 */
public void exampleMethod(int value) {
    // method implementation
}
```

**Comment by shubh220922** [comment]
> all you need to do is change `//CHECKSTYLE:OFF` to `/*CHECKSTYLE:OFF */`

This is what I was trying to say... in ``/*CHECKSTYLE:OFF*/`` the moment I type ``*/`` inside the javadoc (while writing the example), javadoc would think I am terminating it. Image below: 

image
Everything starts to error because javadoc though I was terminating it.

**Comment by shubh220922** [comment]
@romani Done. 
Examples for ``checkC`` and ``checkCPP`` are included.

**Comment by shubh220922** [comment]
Github, generate website

**Comment by shubh220922** [comment]
CI failing. looking into it

**Comment by shubh220922** [comment]
Github, generate website

**Comment by shubh220922** [comment]
Github, generate website

**Review by romani — APPROVED** [review]
Awesome

**Pull request merged**
